File: backend/app/schemas/watchlist.py
from datetime import datetime

from pydantic import BaseModel


class WatchlistAdd(BaseModel):
    ticker: str


class WatchlistItemResponse(BaseModel):
    id: int
    stock_id: int
    ticker: str
    company_name: str
    sector_name: str | None = None
    added_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/watchlist.ts
import apiClient from "./client";

export interface WatchlistItem {
  id: number;
  stock_id: number;
  ticker: string;
  company_name: string;
  sector_name: string | null;
  added_at: string;
}

export async function getWatchlist(): Promise<WatchlistItem[]> {
  const { data } = await apiClient.get("/watchlist");
  return data;
}

export async function addToWatchlist(ticker: string): Promise<WatchlistItem> {
  const { data } = await apiClient.post("/watchlist", { ticker });
  return data;
}

export async function removeFromWatchlist(ticker: string): Promise<void> {
  await apiClient.delete(`/watchlist/${ticker}`);
}
